SuggestedNameAndDir.md

```diff
@@ -0,0 +1,240 @@
+
+The overarching goal here is to provide websites the opportunity to provide the
+file or directory picker implementation suggestions for both the initial
+directory to show in the picker, and (in the case of save dialogs) the file name
+to save a file as. Specifically to help address the following use cases:
+
+### Save an existing file in a different directory
+
+1. User opens a file from their local file system, or from some cloud provider.
+2. User then picks the "Save As" option from the web application.
+3. User saves the file in a different directory, but using the name the file
+   already had (without having to re-type the name).
+
+### Save a file using a different name in the same directory
+
+1. Users opens a file from their local file system.
+2. User then picks the "Save As" option from the web application.
+3. User saves the file using a different name, in the directory the original
+   file exists (without having to re-navigate to that directory).
+
+### Prompt a user to open a image (or video, or audio file)
+
+1. User picks "Insert image" in a web application
+2. A file picker pops up starting out in the users "Pictures" folder.
+
+### Remember different last-used directories for different purposes
+
+1. User opens a document in a web application from directory A.
+2. User inserts an image into the document, by selecting a file from directory B.
+3. User opens a different document, from directory A without having to navigate
+   to directory A again.
+4. Further image insertion operations should have their file pickers start out
+   in the directory of the last image insertion as well.
+
+## Non-goals
+
+All the above functionality is intended as hints/suggestions from websites to
+the file picker implementation to improve the end user experience. However
+implementations should not be required to always follow these suggestions. If
+for example a suggested file name provided by the website is deemed too
+dangerous to be allowed, implementations should be free to ignore the suggestion
+(or adjust the suggested file name to not be dangerous anymore).
+
+## Proposed API
+
+### Specifying suggested file name to save as
+
+```javascript
+const file_ref = await self.showSaveFilePicker({
+  suggestedName: 'README.md',
+  types: [{
+    description: 'Markdown',
+    accept: {
+      'text/markdown': ['.md'],
+    },
+  }],
+});
+```
+
+### Specifying starting directory based on an existing handle.
+
+There are two cases where it is useful to be able to specify a starting
+directory based on an existing handle. The first of these if a website wants to
+let a user pick a sibling to an existing file or directory. For this purpose we
+propose adding a `startInParentOf` option, that can be passed to any of the file
+and directory picker methods:
+
+```javascript
+const existing_handle = /* some FileSystemHandle, could be either a file or a directory */;
+const file_ref = await self.showSaveFilePicker({
+  startInParentOf: existing_handle
```

Comment:
I assume it's out-of-scope to make this recursive. In the example below, if you have a handle on `file_1`, there would be no way to start in `root`. This assumes that the app already as access to `root` from an earlier permission prompt.

```
-root
    |
    subfolder
      |_file_1
      |_file_2
```